backend/main/models/community_users.go

```diff
@@ -98,29 +100,54 @@ func GetUsersForCommunityByType(db *s.Database, communityId, start, count int, u
 	return users, totalUsers, nil
 }
 
-func GetCommunityLeaderboard(db *s.Database, communityId, start, count int) ([]LeaderboardUser, int, error) {
-	var users = []LeaderboardUser{}
-	err := pgxscan.Select(db.Context, db.Conn, &users,
+func GetCommunityLeaderboard(db *s.Database, communityId, start, count int) ([]LeaderboardUserPayload, int, error) {
+	var userAchievements = [] struct {
+		Address   string
+		NumVotes  int
+		EarlyVote int
+	}{}
+	var leaderboardUsers = []LeaderboardUserPayload{}
+	var defaultEarlyVoteWeight = 1
+
+	sql := fmt.Sprintf(
 		`
-		SELECT v.addr, count(*) AS score FROM votes v 
-		JOIN proposals p ON p.id = v.proposal_id
-		WHERE p.community_id = $1
-		GROUP BY v.addr
-		ORDER BY score DESC
-		LIMIT $2 OFFSET $3
-		`, communityId, count, start)
+		SELECT v.addr as address, count(*) as num_votes, 
+			CASE WHEN a.early_vote is NULL THEN 0 ELSE a.early_vote END as early_vote 
+			FROM votes v 
+			LEFT OUTER JOIN proposals p ON p.id = v.proposal_id
+			LEFT OUTER JOIN (
+				SELECT * FROM crosstab(
```

Comment:
This SQL is breaking my brain 😄 

Can you leave a comment roughly describing what is going on here?

Reply:
Done